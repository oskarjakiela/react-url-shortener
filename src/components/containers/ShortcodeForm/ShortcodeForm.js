import React from 'react';

import './ShortcodeForm.css';
import { Button, Input } from 'components/ui';


const displayName = 'Shorty-ShortcodeForm';

const ShortcodeForm = () => (
  <form className={displayName}>
    <Input
      className={`${displayName}__input`}
      placeholder="Paste the link you want to shorten here"
    />

    <Button
      className={`${displayName}__button`}
      disabled
    >
      Shorten this link
    </Button>
  </form>
);

ShortcodeForm.displayName = displayName;

export default ShortcodeForm;
