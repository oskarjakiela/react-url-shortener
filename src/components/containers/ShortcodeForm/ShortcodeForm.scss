.Shorty-ShortcodeForm {
  display: flex;
  margin: 0 -.5rem;
  justify-content: space-between;

  > * {
    margin: 0 .5rem;
  }
}

.Shorty-ShortcodeForm__input {
  flex: 1;
}

.Shorty-ShortcodeForm__button {
}
