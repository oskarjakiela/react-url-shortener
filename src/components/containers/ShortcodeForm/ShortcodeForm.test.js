import React from 'react';
import renderer from 'react-test-renderer';

import ShortcodeForm from './ShortcodeForm';


it('<ShortcodeForm /> renders correctly', () => {
  const tree = renderer
    .create(<ShortcodeForm />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
