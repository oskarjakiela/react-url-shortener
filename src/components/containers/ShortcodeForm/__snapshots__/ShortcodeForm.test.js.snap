// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortcodeForm /> renders correctly 1`] = `
<form
  className="Shorty-ShortcodeForm"
>
  <input
    className="Shorty-Input Shorty-ShortcodeForm__input"
    placeholder="Paste the link you want to shorten here"
  />
  <button
    className="Shorty-Button Shorty-ShortcodeForm__button Shorty-Button--disabled"
    disabled={true}
  >
    Shorten this link
  </button>
</form>
`;
